### PR TITLE
Temporary fix to reenable HIP CI

### PR DESCRIPTION
**Pull request by Rombur**
We have more issues with our AMD machines at ORNL. With this PR, we only compile the tests for the HIP backend but we don't run them. Since we don't run the tests, we can run the CI on the machine we want but we need to provide the architectures ourselves.

### CODE REVIEW

File: .jenkins [hunk 1]
```diff
@@ -135,82 +135,84 @@ pipeline {
...
+                                -DKokkos_ARCH_AMD_GFX90A=ON \
+                                -DKokkos_ENABLE_COMPILER_WARNINGS=ON \
+                                -DKokkos_ENABLE_DEPRECATED_CODE_4=OFF \
+                                -DKokkos_ENABLE_TESTS=ON \
+                                -DKokkos_ENABLE_BENCHMARKS=ON \
+                                -DKokkos_ENABLE_HIP=ON \
+                                -DKokkos_ENABLE_OPENMP=ON \
+                                -DKokkos_ENABLE_HIP_MULTIPLE_KERNEL_INSTANTIATIONS=ON \
+                              .. && \
+                              make -j8'''
```
Comment (dalg24) [inline .jenkins:173]:
```suggestion
                              make -j8'''
//                              make -j8 && ctest --verbose'''
```

File: .jenkins [hunk 2]
```diff
@@ -135,82 +135,84 @@ pipeline {
...
+                                -DKokkos_ARCH_NATIVE=ON \
+                                -DKokkos_ARCH_AMD_GFX90A=ON \
+                                -DKokkos_ENABLE_COMPILER_WARNINGS=ON \
+                                -DKokkos_ENABLE_DEPRECATED_CODE_4=ON \
+                                -DKokkos_ENABLE_DEPRECATION_WARNINGS=OFF \
+                                -DKokkos_ENABLE_TESTS=ON \
+                                -DKokkos_ENABLE_BENCHMARKS=ON \
+                                -DKokkos_ENABLE_HIP=ON \
+                              .. && \
+                              make -j8'''
```
Comment (dalg24) [inline .jenkins:208]:
```suggestion
                              make -j8'''
//                              make -j8 && ctest --verbose'''
```

**Review by dalg24 — APPROVED** [review]

File: .jenkins [hunk 3]
```diff
@@ -135,82 +135,86 @@ pipeline {
...
-//                                -DKokkos_ENABLE_HIP_MULTIPLE_KERNEL_INSTANTIATIONS=ON \
-//                              .. && \
+                stage('HIP-ROCm-5.2') {
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.hipcc'
+                            dir 'scripts/docker'
+                            additionalBuildArgs '--build-arg BASE=rocm/dev-ubuntu-20.04:5.2'
+                            label 'nvidia-docker && ampere'
+                            args '-v /tmp/ccache.kokkos:/tmp/ccache --device=/dev/kfd --device=/dev/dri --security-opt seccomp=unconfined --group-add video --env HIP_VISIBLE_DEVICES=$HIP_VISIBLE_DEVICES'
```
Comment (dalg24) [inline .jenkins:145]: No error with the bogus args?

File: .jenkins [hunk 4]
```diff
@@ -135,82 +135,86 @@ pipeline {
...
-//                                -DKokkos_ENABLE_OPENMP=ON \
-//                                -DKokkos_ENABLE_HIP_MULTIPLE_KERNEL_INSTANTIATIONS=ON \
-//                              .. && \
+                stage('HIP-ROCm-5.2') {
+                    agent {
+                        dockerfile {
+                            filename 'Dockerfile.hipcc'
+                            dir 'scripts/docker'
+                            additionalBuildArgs '--build-arg BASE=rocm/dev-ubuntu-20.04:5.2'
+                            label 'nvidia-docker && ampere'
```
[the remark this answers is not in the document]
Comment (Rombur) [reply .jenkins:144]: No, we also need to provide the architecture instead of using the auto-detection.

File: .jenkins (hunk 3 above)
Comment (dalg24) [reply .jenkins]: Did it give an error?

Comment (Rombur) [reply .jenkins]: The previous try didn't work but there was no error message. Might be that or something else, I don't know.

**Review by crtrott — APPROVED** [review]

**Pull request merged**